src/Action/Delivery/Create.php

```diff
@@ -26,7 +28,20 @@ public function __invoke(Delivery $data)
             throw new ValidationException($errors);
         }
 
-        $this->pricingManager->createOrder($data);
+        $useArbitraryPrice = $this->authorizationCheckerInterface->isGranted('ROLE_ADMIN') && !is_null($data->getDeliveryPriceInput());
+
+        if ($useArbitraryPrice) {
+            $arbitraryPrice = new ArbitraryPrice(
+                $data->getDeliveryPriceInput()->getVariantName(),
+                $data->getDeliveryPriceInput()->getPriceIncVATcents()
+            );
```

Comment:
See other comment, you could avoid doing this conversion by using `ArbitraryPrice` for serialization.